corehq/blobs/s3db.py

```diff
@@ -89,10 +130,19 @@ def _track_transfer(bytes_sent):
             self.metadb.put(meta)
         return meta
 
+    @backoff.on_exception(
```

Comment:
We have a similar thing: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/dimagi/utils/retry.py

That could easily be improved to make the backoff times customizable if necessary. I believe all of the callback parameters here can be done in `should_retry`. Would be nice to not add another dependency.

Reply:
Nice! I didn't know about `retry_on()`. It looks useful, and neat, and a simple way to avoid a dependency. I agree, we'd need different backoff times for S3.

A conversation for another time: When to use a library, and when to roll your own. It's interesting that faced with very similar problems in commcare-hq and commcare-export, two excellent developers chose different options. I wonder if there are heuristics to guide a decision.